packaging/win/delugewin.spec

```diff
@@ -2,30 +2,33 @@
 # -*- mode: python ; coding: utf-8 -*-
 import os
 import sys
-import deluge.common
-from PyInstaller.compat import is_win
-from PyInstaller.utils.hooks import collect_all, copy_metadata
+from PyInstaller.utils.hooks import collect_all, collect_submodules, copy_metadata
 
 datas = []
 binaries = []
-hiddenimports = []
+hiddenimports = ['pygame']
 
 # Collect Meta Data
 datas += copy_metadata('deluge', recursive=True)
 datas += copy_metadata('service-identity', recursive=True)
 
 # Add Deluge Hidden Imports
-tmp_ret = collect_all('deluge')
-hiddenimports += tmp_ret[2]
+hiddenimports += collect_submodules('deluge')
 
-#Add Hidden Imports for Plugins
-tmp_ret2 = collect_all('twisted')
-hiddenimports += tmp_ret2[2]
+# Add Hidden Imports for Plugins
+# Add stdlib as Hidden Imports.
+# This is filtered list that excludes some common examples or stuff not useful in plugins (such as tty, mailbox, tutledemo etc.).
+# It is safe to assume that 90% of that list would already be included anyway.
+if sys.versioninfo.major >= 3 and sys.versioninfo.minor < 10:
+    stdlib = [ 'pickle', 'json', 'email', 'http', 'marshal', 'uuid', 'base64', 'string', 'struct', 'glob', 'ssl', 'urllib', 'html', 'sys',
+      'crypt', 'datetime', 'hmac', 'locale', 'queue', 're', 'enum', 'collections', 'tokenize', 'hashlib', 'xml', 'csb', 'ipaddress', 'os' ]
+if sys.versioninfo.major >= 3 and sys.versioninfo.minor >= 10:
+    for module in sys.stdlib_module_names:
```

Comment:
I know I recommended it myself, but after checking it it seems to pollute resulting binaries by doubling their size. This is most likely due to stuff like `dis`, `compileall`, `mslib`, `winsound`, `pipes`, `trace` and various multimedia/3d modules that are of no use.

I am at the moment conflicted whether we should keep using a list of curated modules or some sort of list of modules we should ignore.

Reply:
The issue with excluding would be that we'd never know if something of Deluge (for example `pygame` doesn't depend on it (for example `winsound`)

Reply:
Also I'd recommend updating the list to a new one I've just made by manually checking modules instead of going by common names:

```
stdlib = [ 'string', 're', 'unicodedata', 'struct', 'codecs', 'datetime', 'zoneinfo', 'calendar', 'collections', 'array', 'weakref', 
  'types', 'copy', 'enum', 'numbers', 'math', 'cmath', 'decimal', 'fractions', 'random', 'statistics', 'itertools', 'functools', 
  'operator', 'pathlib', 'fileinput', 'stat', 'tempfile', 'glob', 'fnmatch', 'shutil', 'pickle', 'copyreg', 'shelve', 'marshal',
  'dom', 'sqlite3', 'zlib', 'gzip', 'bz2', 'lzma', 'csv', 'hashlib', 'hmac', 'secrets', 'os', 'io', 'time', 'logging', 'platform',
  'errno', 'queue', 'socket', 'ssl', 'email', 'json', 'mimetypes', 'base64', 'binhex', 'binascii', 'quopri', 'uu', 'html', 'xml',
  'urllib', 'http', 'ftplib', 'smtplib', 'uuid', 'xmlrpc.client', 'ipaddress', 'locale', 'sys' ]
```

Reply:
curated has worked fine for Deluge in the past so happy to go with that, although nearly all of those are already used by core code anyway. 

Also `csb` looks like typo for `csv`

Reply:
> although nearly all of those are already used by core code anyway.

They are indeed, if I'm not mistaken only `http.cookies`, `urllib.*` and `email.*` was additionally imported (though I've tested on 2.0.5 so that might change with removal of `six` from dependencies)

Reply:
@CirnoT Is this list intended for all python3 versions, not just 3.10?

Reply:
Yes that list works for all Python 3 versions